node-fs-readfile/cat.js

```diff
@@ -0,0 +1,20 @@
+import { readFile } from 'node:fs/promises';
+
+const arrayOfFiles = process.argv.splice(2);
+const promises = arrayOfFiles.map(async (fileName) => {
+  try {
+    const filePath = new URL(fileName, import.meta.url);
+    const content = await readFile(filePath, { encoding: 'utf8' });
+    return content;
+  } catch (error) {
+    console.log(error);
+  }
```

Comment:
Two comments on this. It looks nice and thorough, but it's actually overkill and not necessary.

First, all you are trying to get here is a simple Promise that resolves to the file contents. You don't need to try/catch or async/await here. Promise.all will take care of that.

Second, although the Node documentation may look like you need `import.meta.url`, etc., it actually isn't necessary. By default ,`readFile` will look in the current directory for the file.

So, you can write this much more succinctly as:
`const promises = arrayOfFiles.map((fileName) => readFile(fileName, 'utf8'));`